tests/assembly/dwarf4.rs

```diff
@@ -1,6 +1,6 @@
-// Makes sure that `-Z dwarf-version=4` causes `rustc` to emit DWARF version 4.
+// Makes sure that `-C dwarf-version=4` causes `rustc` to emit DWARF version 4.
 //@ assembly-output: emit-asm
-//@ compile-flags: -g --target x86_64-unknown-linux-gnu -Z dwarf-version=4 -Copt-level=0
+//@ compile-flags: -g --target x86_64-unknown-linux-gnu -C dwarf-version=4 -Copt-level=0
 //@ needs-llvm-components: x86
```

Comment:
Discussion (re. no indication if DWARF is unsupported):

> Currently, passing `-{C,Z} dwarf-version` on targets like `*-windows-msvc` does not do anything. It may be preferable to emit a warning alerting the user that the flag has no effect on the target platform. Alternatively, we could emit an error but this could be annoying since it would require the use of target specific `RUSTFLAGS` to use the flag correctly (and there isn't a way to target "any platform that uses DWARF" using cfgs).

Could we maybe emit an allow-able warn-by-default if `-{C,Z} dwarf-version` is a no-op? It might be surprising or misleading if flag just silently does nothing on unsupported platforms.

Reply:
I think it could make sense to generate a warning or informational message in that case but I could also see it just being annoying on Windows/MSVC where this simply doesn't matter. If the user really *needs* DWARF for some reason, they're going to find out very quickly that DWARF is not used on those targets. For most other deprecated no-opt `-C` flags, we generate an unconditional warning (not a lint).

Reply:
Yeah. It can't be an unconditional warning because you can't silence that. I think maybe a remark about this being no-op in the docs would suffice (at the risk of it becoming outdated, but still)?